src/components/SingleDatePickerInputController.jsx

```diff
@@ -175,7 +175,24 @@ export default class SingleDatePickerInputController extends React.PureComponent
 
   getDisplayFormat() {
     const { displayFormat } = this.props;
-    return typeof displayFormat === 'string' ? displayFormat : displayFormat();
+    
+    if (typeof displayFormat === 'string') {
+      return displayFormat;
+    } else if (Array.isArray(displayFormat)) {
+      return displayFormat[0];
+    }
+    
+    return displayFormat();
+  }
+  
+  getValidationDisplayFormats() {
+    const { displayFormat } = this.props;
+
+    if (typeof displayFormat === 'function') {
+      return displayFormat();
+    }
+    
+    return displayFormat;
```

Comment:
```suggestion
    return [].concat(displayFormat);
```
this way it's always an array